plugins/web-terminal/src/components/TerminalComponent/utils/requests.ts

```diff
@@ -1,67 +1,80 @@
-const NAMESPACE = 'openshift-terminal';
-const WORKSPACE_ENDPOINT = `apis/workspace.devfile.io/v1alpha2/namespaces/${NAMESPACE}/devworkspaces`;
+const WORKSPACE_ENDPOINT = `apis/workspace.devfile.io/v1alpha2/namespaces`;
 
-const workspace = {
-  apiVersion: 'workspace.devfile.io/v1alpha2',
-  kind: 'DevWorkspace',
-  metadata: {
-    generateName: 'web-terminal-',
-    namespace: NAMESPACE,
-    labels: {
-      'console.openshift.io/terminal': 'true',
-    },
-    annotations: {
-      'controller.devfile.io/restricted-access': 'true',
-      'controller.devfile.io/devworkspace-source': 'web-terminal',
+const workspace = (namespace: string) => [
```

Comment:
Can you please explain me why this needs to be an array now? I'm probably missing something and don't see it. It also seems to be called only once here, why don't you inline it into `createWorkspace` function instead?

Reply:
I was trying to inject `namespace` into constant that was not inlined and this was only option that worked :sweat_smile:, but inlining it seems to be far better option in this case so I have changed :+1: .